src/challenge/ChallengeManager.sol

```diff
@@ -110,12 +111,28 @@ contract ChallengeManager is DelegateCallAware, IChallengeManager {
         osp = osp_;
     }
 
-    function postUpgradeInit(IOneStepProofEntry osp_) external onlyDelegated onlyProxyOwner {
-        // when updating to 4844 we need to create new osp contracts and set them here
-        // on the challenge manager
+    /// @dev A osp breaking change is introduced as part of Stylus upgrade, where the new osp would not support
+    ///      pre-Stylus legacy wasmModuleRoot. To ensure that the new osp is not used for legacy wasmModuleRoot,
+    ///      we introduce a conditional OSP where condRoot should be set to the pre-Stylus root and condOsp should
+    ///      be set to the pre-Stylus osp. The correct value should be handled by the upgrade action contract.
+    function postUpgradeInit(
+        IOneStepProofEntry osp_,
+        bytes32 condRoot,
+        IOneStepProofEntry condOsp
+    ) external onlyDelegated onlyProxyOwner {
+        ospCond[condRoot] = condOsp;
```

Comment:
have to do this in postUpgradeInit because proxy owner can only call function in this contract during upgrade